src/uct/ib/rc/accel/rc_mlx5_common.c

```diff
@@ -253,16 +254,17 @@ static inline uct_iface_dm_mp_priv_t* uct_iface_dm_mp_priv(ucs_mpool_t *mp)
 
 static void uct_iface_dm_mp_obj_init(ucs_mpool_t *mp, void *obj, void *chunk)
 {
-    uct_mlx5_dm_data_t *dm         = uct_iface_dm_mp_priv(mp)->dm;
+    uct_mlx5_dm_data_t *dm         = ucs_container_of(mp, uct_mlx5_dm_data_t, mp);
     uct_rc_iface_send_desc_t* desc = (uct_rc_iface_send_desc_t*)obj;
 
-    if ((desc->super.buffer == NULL) && (uct_iface_dm_mp_priv(mp)->count < dm->seg_count)) {
-        desc->lkey          = dm->mr->lkey;
-        desc->super.buffer  = UCS_PTR_BYTE_OFFSET(dm->start_va,
-                                                  uct_iface_dm_mp_priv(mp)->count * dm->seg_len);
-        desc->super.handler = (uct_rc_send_handler_t)ucs_mpool_put;
-        uct_iface_dm_mp_priv(mp)->count++;
-    }
+    ucs_assert(desc->super.buffer == NULL);
+    ucs_assert(uct_iface_dm_mp_priv(mp)->count < dm->seg_count);
```

Comment:
maybe we can remove priv and count, since it's used only for assertion?

Reply:
priv->count is used to assign DM buffer to desc

Reply:
so let's put it in uct_mlx5_dm_data_t?

Reply:
ok, will do

Reply:
done